src/_P101_WakeOnLan.ino

```diff
@@ -0,0 +1,475 @@
+          parse_error = true;
+          msgStr      = wolStr;
+          msgStr     += F("Error, IP Addr Invalid [");
+          msgStr     += paramIp;
+          msgStr     += F("]");
+          addLog(LOG_LEVEL_INFO, msgStr);
+        }
+
+        // Validate UDP Port.
+        if (!validatePort(paramPort)) {
+          parse_error = true;
+          msgStr      = wolStr;
+          msgStr     += F("Error, Port Invalid [");
+          msgStr     += paramPort;
+          msgStr     += F("]");
+          addLog(LOG_LEVEL_INFO, msgStr);
+        }
+
+        // If no errors we can send Magic Packet.
+        if (parse_error == true) {
+          msgStr = F("CMD Syntax Error");
+          addLog(LOG_LEVEL_INFO, wolStr + msgStr);
+          msgStr += F(" <br>");
+          SendStatus(event->Source, msgStr); // Reply (echo) to sender. This will print message on browser.
+        }
+        else {                               // No parsing errors, Send Magic Packet (Wake Up the MAC).
+          msgStr  = wolStr;
+          msgStr += F("MAC= ");
+          msgStr += paramMac;
+          msgStr += F(", IP= ");
+          msgStr += paramIp;
+          msgStr += F(", Port= ");
+          msgStr += paramPort;
+          addLog(LOG_LEVEL_INFO, msgStr);
+
+          // Send Magic Packet.
+          if (WiFi.status() == WL_CONNECTED) {
+            IPAddress local_IP;
+            local_IP.fromString(paramIp);
+            WOL.setBroadcastAddress(local_IP);
+
+            // WOL.setRepeat(1, 0); // One Magic Packet, No Repeats. (Library default)
+
+            if (!WOL.sendMagicPacket(paramMac, paramPort.toInt())) {
+              msgStr  = wolStr;
+              msgStr += F("Error, Magic Packet Failed (check parameters)");
+              addLog(LOG_LEVEL_INFO, msgStr);
+            }
+          }
+          else {
+            msgStr  = wolStr;
+            msgStr += F("Error, WiFi Off-Line");
+            addLog(LOG_LEVEL_INFO, msgStr);
+          }
+        }
+      }
+      break;
+    }
+
+    case PLUGIN_EXIT:
+    {
+      break;
+    }
+  }
+  return success;
+}
+
+// ************************************************************************************************
+// validateIp(): IPv4 Address validity checker.
+// Arg: IP Address String in dot separated format (192.168.1.255).
+// Return true if IP string appears legit.
+bool validateIp(String ipStr) {
+  IPAddress ip;
+  unsigned int length = ipStr.length();
+
+  if ((length < IP_MIN_SIZE_P101) || (length > IP_ADDR_SIZE_P101)) {
+    return false;
+  }
+  else if (ip.fromString(ipStr) == false) { // ThomasTech's Trick to Check IP for valid formatting.
+    return false;
+  }
+
+  return true;
+}
+
+// ************************************************************************************************
+// validateMac(): MAC Address validity checker.
+// Arg: MAC Address String in colon separated format (00:00:00:00:00:00).
+// Return true if MAC string appears legit.
+bool validateMac(String macStr) {
```

Comment:
On some OS'es the leading 0 in a hex field may be left out.  (e.g. Mac OS)
Copy-paste of those MAC addresses will thus be rejected.

Reply:
During submission, the user will see a message in the plugin that reports any problems with the MAC formatting. This keeps things simple. 

So if a copied MAC string is not formatted as aa:bb:cc:dd:ee:ff then the user will be warned. And will need to manually add the missing or illegal chars and re-submit.